hnn_core/params.py

```diff
@@ -140,14 +140,17 @@ def _extract_bias_specs_from_hnn_params(params, cellname_list):
     return bias_specs
 
 
-def _extract_drive_specs_from_hnn_params(params, cellname_list):
+def _extract_drive_specs_from_hnn_params(
+        params, cellname_list, legacy_mode=False):
```

Comment:
should we call this function with `legacy_mode=False` also in `add_drives_from_params` to simplify the logic there and contain it in this function?